packages/webpack-extraction-plugin/README.md

```diff
@@ -1,3 +1,71 @@
-# TBD
+# Webpack plugin to perform CSS extraction in Griffel
 
-Not ready for consuming.
+> ⚠️ **Note** This package is experimental, API may change.
+
+A plugin for Webpack 5 that performs CSS extraction for [`@griffel/react`](../react).
+
+## Install
+
+```bash
+yarn add --dev @griffel/webpack-extraction-plugin
+# or
+npm install --dev @griffel/webpack-extraction-plugin
+```
+
+## When to use it?
+
+It's designed to be used only in applications.
+
+## How it works?
+
+This plugin relies on assets transformed by [`@griffel/babel-preset`](../babel-preset) or [`@griffel/webpack-loader`](../webpack-loader). Usage of these utilities is a **prerequisite**.
+
+The plugin transforms code to remove generated CSS from JavaScript files and create CSS assets.
+
+_Currently, all CSS rules will be extracted to a single CSS file i.e. [code splitting](https://webpack.js.org/guides/code-splitting/) for extracted CSS **will not work**._
+
+## Usage
+
+Webpack documentation:
+- [Loaders](https://webpack.js.org/loaders/)
+- [Plugins](https://webpack.js.org/concepts/plugins/)
+
+> Please configure [`@griffel/webpack-loader`](../webpack-loader) first.
+
+Within your Webpack configuration object, you'll need to add the loader and the plugin from `@griffel/webpack-extraction-plugin` like so:
+
+```js
+const { GriffelCSSExtractionPlugin } = require('@griffel/webpack-extraction-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.(js|ts|tsx)$/,
+        // Apply "exclude" only if your dependencies **do not use** Griffel
```

Comment:
should `include` be added to the docs too?

Reply:
Added a section below 64a52cd